### ISSUE TITLE
Merge Layer issue

### ISSUE THREAD

**Issue by don570** [mypaint/mypaint#1005]
I noticed another  issue  with 'Merge  layer  down'
when merging  a normal  layer  down  to a pigment  layer.
Instead  of  pink  skin  I see yellow...
I include a painting  as an example.

1. Start MyPaint
2. Open my two layer  painting
shareable link
https://drive.google.com/open?id=1mOi5XO23wQTm4rX_Xj7X8HJl_T1YE82e

3. Layer > Merge layer down
4.   Pink  skin  has  changed  color  (to yellow)
_____________________________________________________

**Comment by briend** [comment]
This and #992 have been really troubling me.  I really felt like I was missing something obvious, but I finally tried similar operations in Krita and Gimp noticed the same kinds of issues.  Thanks for reporting these, though!

Pigment mode is a lot like the "Geometric Mean" layer mode in Krita.  Try that and various other "exotic" layer modes in Krita along with merge down operations.  You'll notice a lot of breakage.  GIMP has some issues as well I believe  ( tried 2.8 w/ a multiply mode and there was breakage).

I've tried to make MyPaint behave as rationally as I can think.  Here's how merging two layers should work in MyPaint right now:

Pigment merged with Pigment == one layer, Pigment mode (100% correct looking)
Normal or other non-Pigment modes merged with Normal == one layer, Normal mode (100% correct looking)
Pigment merged with Normal == one layer, Pigment Mode (bottom layer switched to pigment, looking darker/different)==broken.
Normal merged with Pigment == one layer, Normal Mode (bottom layer switched to normal, looking lighter/different)==broken.

So for now I would highly recommend not trying to merge Pigment mode with anything except Pigment mode layers.  I'm thinking about disabling the option to even try, but then again GIMP and Krita allow you to perform these sorta buggy operations, so maybe let the user decide?

I believe the source of the problem is that the effect of the background has to be "subtracted" from the resulting merged layers.  However, for non-linear modes it is not a simple linear subtraction of the background.  There has to be some kind of "solver" to find a solution where the result looks the same but the layer mode is now fundamentally different (linear vs non-linear).  If it was simple I think GIMP and Krita might have figured it out by now :-)
